src/service-providers/common/common.adapters.ts

```diff
@@ -9,3 +9,17 @@ export const adaptFilteringStatsCounts = (
     transcripts: counts[2],
   }
 }
+
+export const adaptDataToCamelizedType = <T>(response: T): T => {
```

Comment:
I think this generic adapter is not good solution, because information about types are lost(

Reply:
Also you converts only first level keys of object

Reply:
I think we can save original API types with kebab-cases and create new adapted type with camelized keys.
With some type utilities, e.g.:
```
type KebabToCamel<T extends string> = T extends `${infer A}-${infer B}`
  ? `${A}${Capitalize<B>}`
  : T

type KebabToCamelObject<T> = T extends Record<string, unknown>
  ? {
      [K in keyof T as KebabToCamel<K & string>]: KebabToCamelObject<T[K]>
    }
  : T extends Array<infer U>
  ? Array<KebabToCamelObject<U>>
  : T

type Foo = {
  'asd-qwe': number
  'foo-bar': {
    'abc-zyx': boolean
  }
  'some-array': {
    'some-key': string
  }[]
}

type Bar = KebabToCamelObject<Foo>

const z: Bar = {
  asdQwe: 2,
  fooBar: {
    abcZyx: true,
  },
  someArray: [
    {
      someKey: 'qwe',
    },
    {
      someKey: 'asd',
    },
  ],
}
```

Reply:
>I think this generic adapter is not good solution
Why not, it might be used in various places by piping, for example

Reply:
>I think we can save original API types

Do we have cases when we need original types?
The only case comes up on mind is when we receive some piece of data, do something with it and push back on the server with the exact same interface. But I can't remember any cases that follow that pattern.

Reply:
>Also you converts only first level keys of object

Agree, converting must be recursive

Reply:
added recursive camelizator